rustdoc: remove CSS `@media (min-width: 701px)`

notriddle opened the pull request:
The two rules within it can and should be done without the separate media query:

  * There ain't no rule saying a viewport can't be `700.5px` wide, since hardware pixels can be finer than CSS pixels.

    <details><summary>Screenshot</summary>

    ![image](https://user-images.githubusercontent.com/1593513/196283804-e946fbe3-d914-4819-ba2d-b4a6a07f190a.png)

    </details>

  * The rule for the first example-wrap child should probably apply on mobile.
    
    <details><summary>Screenshots</summary>

    ## Before

    ![rustdoc-before](https://user-images.githubusercontent.com/1593513/196282813-88b861c7-7b05-4de5-a421-914396f86a04.png)

    ## After

    ![rustdoc-after](https://user-images.githubusercontent.com/1593513/196282842-e57760c2-72d5-4dad-94f5-1d7175043326.png)

    </details>

  * The rule for the source sidebar is overriden by the mobile rule setting `max-width: 100vw`, so it can be merged with the rest of the styles.

notriddle committed:
rustdoc: remove CSS `@media (min-width: 701px)`

The two rules within it can and should be done without the separate
media query:

  * There ain't no rule saying a viewport can't be `700.5px` wide, since
    hardware pixels can be finer than CSS pixels.

  * The rule for the first example-wrap child should probably apply
    on mobile.

  * The rule for the source sidebar is overriden by the mobile rule
    setting `max-width: 100vw`, so it can be merged with the rest
    of the styles.

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @jsha (or someone else) soon.

Please see [the contribution instructions](https://rustc-dev-guide.rust-lang.org/contributing.html) for more information.

notriddle committed:
rustdoc: change mobile breakpoint to match CSS

Since it's possible to have a 700.5px viewport width, the JS needs to not
switch to mobile mode in such a setup.

GuillaumeGomez commented on src/test/rustdoc-gui/sidebar-mobile-scroll.goml:
```diff
@@ -1,12 +1,12 @@
 // Scroll down.
 scroll-to: "//h2[@id='blanket-implementations']"
-assert-window-property: {"pageYOffset": "651"}
+assert-window-property: {"pageYOffset": "627"}
```
I can add one. :)

GuillaumeGomez replied:
Opened https://github.com/GuillaumeGomez/browser-UI-test/issues/394.

GuillaumeGomez commented:
Thanks!

@bors r+ rollup

bors commented:
:pushpin: Commit d4e4b08b9b53f3dda7be3cc3bf8881db4180f736 has been approved by `GuillaumeGomez`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=GuillaumeGomez d4e4b08b9b53f3dda7be3cc3bf8881db4180f736 -->
<!-- homu: {"type":"Approved","sha":"d4e4b08b9b53f3dda7be3cc3bf8881db4180f736","approver":"GuillaumeGomez","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
